### PR TITLE
feat(infinite-hits): add banners to vue flavor

### PR DESCRIPTION
**Pull request by taylorcjohnson**
**Summary**

This PR adds the new top banner to the Vue `InfiniteHits` widget and updates associated tests. The top banner feature has already been added to `Hits` and `InfiniteHits` for the JS and React flavors of InstantSearch.

This widget is slightly different from the React version by using a `showBanner` prop (defaulted to `true`) since relying on whether a parent component passed included content with the banner slot could be difficult. I'm open to changing this implementation if anyone has a recommended approach though.

[Jira issue](https://algolia.atlassian.net/browse/EMERCH-1560)

**Result**

Getting Started example app:
image

### CODE REVIEW

File: tests/common/widgets/infinite-hits/options.ts [hunk 1]
```diff
@@ -108,88 +106,86 @@ export function createOptionsTests(
...
-            widgets: {
-              banners: [
-                {
-                  image: {
-                    urls: [{ url: 'https://via.placeholder.com/550x250' }],
-                  },
-                  link: {
-                    url: 'https://www.algolia.com',
-                  },
+    test('renders default banner element with banner widget renderingContent', async () => {
```
Comment (taylorcjohnson) [inline tests/common/widgets/infinite-hits/options.ts:109]: The snapshot output for the Vue version of this test is adding an extra line between the banner and hit list, but I cannot figure out why. I must be missing something...

https://app.circleci.com/pipelines/github/algolia/instantsearch/11178/workflows/90ab6bc7-58a9-4185-94ab-18d1594139c4/jobs/54699

Comment (dhayab) [reply tests/common/widgets/infinite-hits/options.ts:109]: Could this be an undesirable side-effect of prettier applying rules when saving, which might break the generated snapshot? If so, you can either update the snapshot without saving the file again from your editor, or save without formatting being applied.

Comment (taylorcjohnson) [reply tests/common/widgets/infinite-hits/options.ts:109]: I will double-check, but I don't believe so. The test passes for JS and React, but fails only for Vue.

Comment (dhayab) [reply tests/common/widgets/infinite-hits/options.ts:109]: Ah, this then means that the Vue serializer used for Jest outputs the DOM differently then the other ones. To "fix" this, we smooth out differences like this in the normalize function that should be at the top of the file, by usually adding a regular expression that targets the relevant extra whitespace and removes it.

Comment (taylorcjohnson) [reply tests/common/widgets/infinite-hits/options.ts:111]: Yep, @aymeric-giraudet had just sent me a snippet to try. Worked locally, just pushed another commit.

**Review by dhayab — APPROVED** [review]

**Review by aymeric-giraudet — APPROVED** [review]
I think bundlesize threshold needs to be updated too, otherwise looks good !

**Pull request merged**